java-checks/src/main/java/org/sonar/java/checks/AssignmentInSubExpressionCheck.java

```diff
@@ -58,23 +59,27 @@ public void visitAnnotation(AnnotationTree annotationTree) {
 
   @Override
   public void visitLambdaExpression(LambdaExpressionTree lambdaExpressionTree) {
-    //skip lambda if body is an assignement
+    //skip lambda if body is an assignment
     if(!lambdaExpressionTree.body().is(Kind.ASSIGNMENT)) {
```

Comment:
The body can be assignment to a field, which is allowed to not be final even if used in lambda. But I agree this is not at all obvious, I had to google the expression to find old thread on ML http://sonarqube-archive.15.x6.nabble.com/Assignment-within-sub-expression-rule-for-lambdas-td5029472.html 

I made another commit making this test more explicit